code/modules/clothing/masks/gasmask.dm

```diff
@@ -13,14 +13,54 @@
 
 /obj/item/clothing/mask/gas/atmos
 	name = "atmospheric gas mask"
-	desc = "Improved gas mask utilized by atmospheric technicians. Still not very good at blocking gas flow, but it's flameproof!"
+	desc = "Improved gas mask utilized by atmospheric technicians. It's flameproof!"
 	icon_state = "gas_atmos"
 	inhand_icon_state = "gas_atmos"
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS | GAS_FILTERING
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 10, FIRE = 20, ACID = 10)
 	w_class = WEIGHT_CLASS_SMALL
 	gas_transfer_coefficient = 0.001 //cargo cult time, this var does nothing but just in case someone actually makes it do something
 	permeability_coefficient = 0.001
 	resistance_flags = FIRE_PROOF
+	var/max_filters = 3
+	var/current_filters = 0
+	var/list/gas_filters
+
+/obj/item/clothing/mask/gas/atmos/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The amount of slots for filters are [max_filters].</span>"
+	if(current_filters > 0)
+		. += "<span class='notice'>Currently there are [current_filters] filters with [get_filter_durability()]% durability.</span>"
+
+/obj/item/clothing/mask/gas/atmos/attackby(obj/item/I, mob/user)
+	. = ..()
+	if(!istype(I, /obj/item/gas_filter))
+		return TRUE
+	if(current_filters >= max_filters)
+		return TRUE
+	if(!user.transferItemToLoc(I, src))
+		return TRUE
+	current_filters++
+	LAZYADD(gas_filters, I)
+
+/obj/item/clothing/mask/gas/atmos/consume_filter()
+	if(current_filters <= 0)
+		return FALSE
+	var/obj/item/gas_filter/gas_filter = pick(gas_filters)
+	gas_filter.reduce_filter_status()
+	if(gas_filter.filter_status <= 0)
+		current_filters--
+		LAZYREMOVE(gas_filters, gas_filter)
+		qdel(gas_filter)
+	return TRUE
+
+/obj/item/clothing/mask/gas/atmos/proc/get_filter_durability()
+	var/max_filters_durability = current_filters * 100
+	var/current_filters_durability
+	for(var/obj/item/gas_filter/gas_filter in gas_filters)
+		current_filters_durability += gas_filter.filter_status
+	var/durability = (current_filters_durability / max_filters_durability) * 100
+	return durability
```

Comment:
all of this stuff above should be put on the normal gas mask, especially so the emergency mask isn’t a snowflaked atmos mask subtype

Reply:
i'll see if it can be done easily, because gas masks comprehend many things, like clown and mime masks

Reply:
for things like those, set the total amount of filters to zero or something. do note that both of those are meant to be their emergency masks

Reply:
ok i made that the procs are in the `/obj/item/clothing/mask/gas` level and then i just set the max_filters to zero to most of the masks